### PR TITLE
python3Packages.torchmetrics: 1.8.2 -> 1.9.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

Diff: https://github.com/Lightning-AI/torchmetrics/compare/v1.8.2...v1.9.0
Changelog: https://github.com/Lightning-AI/torchmetrics/releases/tag/v1.9.0

cc @SomeoneSerge

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

### CODE REVIEW

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 497894`
Commit: `d4628d9dd5cf3df2398f8445487c8e1771ae2dfc`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 29 packages built:</summary>
  <ul>
    <li>calibre</li>
    <li>convertx</li>
    <li>piper-tts</li>
    <li>piper-tts.dist</li>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>python314Packages.fastmri</li>
    <li>python314Packages.fastmri.dist</li>
    <li>python314Packages.finetuning-scheduler</li>
    <li>python314Packages.finetuning-scheduler.dist</li>
    <li>python314Packages.lightning</li>
    <li>python314Packages.lightning.dist</li>
    <li>python314Packages.pytorch-lightning</li>
    <li>python314Packages.pytorch-lightning.dist</li>
    <li>python314Packages.torchmetrics</li>
    <li>python314Packages.torchmetrics.dist</li>
    <li>unbook</li>
    <li>wyoming-piper</li>
    <li>wyoming-piper.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 29 packages built:</summary>
  <ul>
    <li>calibre</li>
    <li>convertx</li>
    <li>piper-tts</li>
    <li>piper-tts.dist</li>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>python314Packages.fastmri</li>
    <li>python314Packages.fastmri.dist</li>
    <li>python314Packages.finetuning-scheduler</li>
    <li>python314Packages.finetuning-scheduler.dist</li>
    <li>python314Packages.lightning</li>
    <li>python314Packages.lightning.dist</li>
    <li>python314Packages.pytorch-lightning</li>
    <li>python314Packages.pytorch-lightning.dist</li>
    <li>python314Packages.torchmetrics</li>
    <li>python314Packages.torchmetrics.dist</li>
    <li>unbook</li>
    <li>wyoming-piper</li>
    <li>wyoming-piper.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>piper-tts</li>
    <li>piper-tts.dist</li>
    <li>wyoming-piper</li>
    <li>wyoming-piper.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>python314Packages.fastmri</li>
    <li>python314Packages.fastmri.dist</li>
    <li>python314Packages.finetuning-scheduler</li>
    <li>python314Packages.finetuning-scheduler.dist</li>
    <li>python314Packages.lightning</li>
    <li>python314Packages.lightning.dist</li>
    <li>python314Packages.pytorch-lightning</li>
    <li>python314Packages.pytorch-lightning.dist</li>
    <li>python314Packages.torchmetrics</li>
    <li>python314Packages.torchmetrics.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>piper-tts</li>
    <li>piper-tts.dist</li>
    <li>wyoming-piper</li>
    <li>wyoming-piper.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 22 packages built:</summary>
  <ul>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>python314Packages.fastmri</li>
    <li>python314Packages.fastmri.dist</li>
    <li>python314Packages.finetuning-scheduler</li>
    <li>python314Packages.finetuning-scheduler.dist</li>
    <li>python314Packages.lightning</li>
    <li>python314Packages.lightning.dist</li>
    <li>python314Packages.pytorch-lightning</li>
    <li>python314Packages.pytorch-lightning.dist</li>
    <li>python314Packages.torchmetrics</li>
    <li>python314Packages.torchmetrics.dist</li>
  </ul>
</details>

**Review by kirillrdy — APPROVED** [review]
